src/export/ExportWavPack.cpp

```diff
@@ -371,21 +413,39 @@ ProgressResult ExportWavPack::Export(AudacityProject *project,
             n = wxT("DATE");
          }
          WavpackAppendTagItem(wpc,
-                              (char *) (const char *) n.mb_str(wxConvUTF8),
-                              (char *) (const char *) v.mb_str(wxConvUTF8),
-                              (int) v.length());
+                              n.mb_str(wxConvUTF8),
+                              v.mb_str(wxConvUTF8),
+                              static_cast<int>( v.length() ));
       }
 
       if (!WavpackWriteTag(wpc)) {
-         // Not sure what to do
+         ShowExportErrorDialog( WavpackGetErrorMessage(wpc) );
+         return ProgressResult::Failed;
       }
    }
 
-   WavpackCloseFile(wpc);
-
    if ( !outWvFile.file.get()->Close()
       || ( outWvcFile.file && outWvcFile.file.get() && !outWvcFile.file.get()->Close())) {
-      updateResult = ProgressResult::Cancelled;
+      return ProgressResult::Failed;
+   }
+
+   // wxFile::Create opens the file with only write access
+   // So, need to open the file again with both read and write access
+   if (!outWvFile.file->Open(fName.GetFullPath(), wxFile::read_write)) {
```

Comment:
For simplicity we can cache that block in memory, but this will do too. Very good!